ha/setup/v2/ha_setup.py

```diff
@@ -0,0 +1,233 @@
+from ha.execute import SimpleCommand
+from ha import const
+
+class Cmd:
+
+    """Setup Command"""
+    _index = "conf"
+
+    def __init__(self, args: dict):
+        """Init method."""
+        self._url = args.config
+        Conf.load(self._index, self._url)
+        self._args = args.args
+        self._execute = SimpleCommand()
+
+    @property
+    def args(self) -> str:
+        return self._args
+
+    @property
+    def url(self) -> str:
+        return self._url
+
+    @staticmethod
+    def usage(prog: str):
+        """Print usage instructions"""
+        sys.stderr.write(
+            f"usage: {prog} [-h] <cmd> <--config url> [--plan <TYPE>] <args>...\n"
+            f"where:\n"
+            f"cmd   post_install, config, init, test, upgrade, reset, cleanup, backup, restore\n"
+            f"--config   Config URL\n"
+            f"--plan   Test plan")
+
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """Return the Command after parsing the command line."""
+        parser = argparse.ArgumentParser(desc)
+
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+        args = parser.parse_args(argv)
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """Add Command args for parsing."""
+        parser1 = parser.add_parser(cls.name, help='setup %s' % name)
+        parser1.add_argument('--config', help='Config URL')
+        parser1.add_argument('--plan', nargs="?", default=[], help='Test plan')
+        parser1.add_argument('args', nargs='*', default=[], help='args')
+        parser1.set_defaults(command=cls)
+
+class PostInstallCmd(Cmd):
+
+    """PostInstall Setup Cmd"""
+    name = "post_install"
+    packages = ["pacemaker", "corosync", "pcs"]
+
+    def __init__(self, args: dict):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        # Pre-requisite checks are done here.
+        # Make sure the pacemaker, corosync and pcs packages have been installed
+        PkgV().validate('rpms', self.packages)
+
+class ConfigCmd(Cmd):
+
+    """Setup Config Cmd"""
+    name = "config"
+
+    def __init__(self, args):
+        """Init method."""
+        super().__init__(args)
+
+    def process(self):
+        # Read machine-id and using machine-id read node name from confstore
+        # This node name will be used for adding the node to the cluster.
+        command = "cat /etc/machine-id"
+        machine_id, err, rc = self._execute.run_cmd(command, check_error=True)
+        Log.info(f"Read machine-id. Output: {machine_id}, Err: {err}, RC: {rc}")
+        node_name = Conf.get(self._index, f"cluster.server_nodes.{machine_id.strip()}")
```

Comment:
is it host name or minion name

Reply:
I think a minion name e.g. srvnode_1, I will rename the local variable accordingly.